Mathlib/Data/Fintype/Pi.lean

```diff
@@ -133,3 +135,26 @@ lemma Fin.snoc_mem_piFinset_snoc_iff {n : ℕ} {α : Fin (n + 1) → Type*}
     (Sᵢ : (i : Fin n) → Finset (α i.castSucc)) (Sₙ : Finset (α <| .last n)) :
     Fin.snoc xs x ∈ Fintype.piFinset (Fin.snoc Sᵢ Sₙ) ↔ xs ∈ Fintype.piFinset Sᵢ ∧ x ∈ Sₙ := by
   simp_rw [Fin.mem_piFinset_succ_iff', init_snoc, snoc_last]
+
+-- TODO Generalize from `succ` to `succAbove`. Requires `Fin.forall_fin_succAbove`.
+lemma Finset.map_piFinSuccAboveEquiv_filter_piFinset {n : ℕ} {α : Fin (n + 1) → Type*}
+    (p : ((i : Fin n) → α i.succ) → Prop) [DecidablePred p]
+    (S : (i : Fin (n + 1)) → Finset (α i)) :
+    Finset.map
+      ((Equiv.piFinSuccAboveEquiv α 0).toEmbedding)
+      (Finset.filter (fun r ↦ p (fun x ↦ r $ Fin.succ x)) (Fintype.piFinset S))
+      = ((S 0) ×ˢ Finset.filter p (Fintype.piFinset (fun x ↦ S $ Fin.succ x))) := by
```

Comment:
Can you add a copy of this with `Equiv.symm` on the other side? There should be some lemma that lets you move the `map` across, so that you don't have to prove it again from scratch.

Reply:
Ok I added `Finset.filter_piFinset_eq_map_piFinSuccAboveEquiv_symm`, lmk if that's not what you were looking for.